Songbird

melk0sha opened the pull request:
1. Task: [task link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/songbird.md)
2. Screenshot: [screenshot link](https://ibb.co/vcSN63j)
3. Deploy: https://melk0sha-songbird.netlify.com/
4. 05.02.2020 / 12.02.2020
5. Total:

Task (210 / 230)

**Вёрстка, дизайн, UI**
_+30:_ минимальная ширина страницы, при которой приложение отображается корректно – 320 рх

**Аудиоплеер**
стандартный
_+20:_ кастомный
для кастомизации плеера можно использовать библиотеки

**Контент**
_+0:_ Использование готового файла с данными birds data
Создание и использование собственного файла с данными

**Header**
_+10:_ правильное отображение текущего счета игры
_+10:_ индикация текущего вопроса

**Блок с вопросом**
_+10:_ подстановка дефолтного изображения и замена названия птицы на символы (*), пока игрок не выберет правильный ответ
_+10:_ при выборе правильного ответа появляется изображение и название птицы

**Блок с вариантами ответов (названия птиц)**
_+20:_ цветовая индикация правильного/неправильного ответа
_+10:_ при клике по названию птицы в блоке с описанием птицы отображается информацию о ней
Если правильный ответ уже дан, возможность просматривать описания птиц при клике по названию остаётся, цвет индикаторов при этом не изменяется.

**Блок с описанием птицы**
_+10:_ пока игрок не кликнул по названию птицы из списка, в блоке выводится короткий текст с предложением послушать плеер и выбрать название птицы, чей голос прозвучал
_+20:_ при клике по названию птицы из списка, в блоке с описанием птицы появляется актуальная информация о ней
Информация о птице включает:
- изображение
- название (на русском и на латыни)
- аудиоплеер с записью голоса
- дополнительное описание.

**Кнопка перехода к следующему вопросу**
_+10:_ пока не выбран правильный ответ, кнопка не активна, нет возможности перейти к следующему заданию. Активное и неактивное состояние кнопки визуально отличаются, например, активная кнопка имеет зеленый, не активная - серый цвет
_+20:_ после правильного ответа на последний вопрос, клик по кнопке скрывает блок с вопросом, блок с вариантами ответов и блок с описанием птицы. Появляется сообщение с поздравлением об окончании игры
Сообщение содержит количество набранных баллов из максимально возможных и кнопку с предложением сыграть ещё раз.

**Extra scope**
_+20:_ звуковая индикация правильного/неправильного ответа
- при выборе правильного или неправильного ответа издаются разные звуковые сигналы
- при выборе неправильного ответа проигрывание аудиоплеера не должно останавливаться
- при выборе правильно ответа аудиоплеер должен остановиться

_+10:_ Поздравление с абсолютной победой, если набрано максимально возможное количество баллов
Здесь можно проявить свою фантазию. Это может быть все, что угодно. Например: **картинка**, анимация, видео, ссылка на что-то интересное, **хорошие слова**, сертификат знатока птичьих голосов, слоган "теперь ты знаешь кто чирикнул" :)

**Штрафы**
_-0:_ ошибки в консоли, связанные с работой приложения до -10
_-0:_ любой сбой/поломка приложения, связанная с действиями пользователя до -20

Mark calculation: 230 - 20 = 210

melk0sha committed:
init: project start

melk0sha committed:
feat: add data and main js file

melk0sha committed:
feat: add assets

melk0sha committed:
feat: add components with rendering

melk0sha committed:
feat: add styles

melk0sha committed:
feat: add options for projects' needs

melk0sha committed:
feat: add and change components

melk0sha committed:
feat: add congrats img

melk0sha committed:
feat: make responsive layout

melk0sha committed:
feat: add favicon

melk0sha committed:
feat: add depends

melk0sha committed:
refactor and feat: add sounds and refactor code style

melk0sha committed:
fix: audio stopping

Alexshumansky commented on src/components/App.js:
```diff
@@ -0,0 +1,161 @@
+  constructor() {
+    super();
+
+    this.minPointOfBirdCount = 0;
+    this.maxPointOfBirdCount = 5;
+    const current = getCurrentNumberOfBird(
+      this.minPointOfBirdCount,
+      this.maxPointOfBirdCount
+    );
```
this.minPointOfBirdCount = 0; this.maxPointOfBirdCount = 5; - лишние магические числа, и в в this не надо их пихать, можно вовнутрь эти 0 и 5 пихнуть либо завести папку constants, куда ты напишешь все свои константы для нужного компонента, а лучше взять для гибкости data[0].length

Alexshumansky commented on src/components/header/Header.js:
```diff
@@ -0,0 +1,17 @@
+      <header>
+        <div className="top-header">
+          <div className="logo"></div>
+          <Score score={this.props.score} />
+        </div>
+        <BirdIndicator selected={this.props.selected} />
+      </header>
```
деструктуризировать пропсы везде, не только тут

Alexshumansky commented on src/components/header/BirdIndicatorItem.js:
```diff
@@ -0,0 +1,10 @@
+
+export default class BirdIndicatorItem extends Component {
+  render() {
+    const classes = ["bird-type"];
+    this.props.selected ? classes.push("selected") : null;
+
+    return <li className={classes.join(" ")}>{this.props.type}</li>;
```
деструктуризация
const classes = 'bird-type';
classes += "selected"; Зачем тут массив?

Alexshumansky commented on src/components/header/BirdIndicator.js:
```diff
@@ -0,0 +1,27 @@
+
+export default class BirdIndicator extends Component {
+  render() {
+    const types = [
+      "Разминка",
+      "Воробьиные",
+      "Лесные птицы",
+      "Певчие птицы",
+      "Хищные птицы",
+      "Морские птицы"
+    ];
```
это всё добро в константу, в папку constants ну или локально, но это хуже, и каждому табу дай id, в качестве key использовать его, завязывать key на id - плохая практика и не имеет никакого смысла

Alexshumansky commented on src/components/header/BirdIndicator.js:
```diff
@@ -0,0 +1,27 @@
+      <ul className="bottom-header">
+        {types.map((type, idx) =>
+          this.props.selected === idx ? (
+            <BirdIndicatorItem key={idx} type={type} selected={true} />
+          ) : (
+            <BirdIndicatorItem key={idx} type={type} selected={false} />
```
key, см.в.
id + selected можно прокинуть дальше в BirdIndicatorItem и внутри делать что-то аля
selected === id ? classes.push..., и тогда избавишься от ифа здесь и лишнего пропса selected
получится чище
types.map((type) => <BirdIndicatorItem key={type.id} selected={this.props.selected} type={type}  />

melk0sha committed:
docs: update README

melk0sha committed:
docs: fix text in README

melk0sha committed:
docs: update linebreak

melk0sha committed:
docs: update headers size

dependabot[bot] committed:
build(deps): bump lodash from 4.17.15 to 4.17.19

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump elliptic from 6.5.2 to 6.5.3

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.2 to 6.5.3.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.2...v6.5.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump http-proxy from 1.18.0 to 1.18.1

Bumps [http-proxy](https://github.com/http-party/node-http-proxy) from 1.18.0 to 1.18.1.
- [Release notes](https://github.com/http-party/node-http-proxy/releases)
- [Changelog](https://github.com/http-party/node-http-proxy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/http-party/node-http-proxy/compare/1.18.0...1.18.1)

Signed-off-by: dependabot[bot] <support@github.com>

melk0sha committed:
Merge pull request #6 from melk0sha/dependabot/npm_and_yarn/http-proxy-1.18.1

build(deps): bump http-proxy from 1.18.0 to 1.18.1

melk0sha committed:
Merge pull request #4 from melk0sha/dependabot/npm_and_yarn/lodash-4.17.19

build(deps): bump lodash from 4.17.15 to 4.17.19

melk0sha committed:
Merge pull request #5 from melk0sha/dependabot/npm_and_yarn/elliptic-6.5.3

build(deps): bump elliptic from 6.5.2 to 6.5.3
